envoy/stats/allocator.h

```diff
@@ -58,6 +58,28 @@ class Allocator {
   virtual const SymbolTable& constSymbolTable() const PURE;
   virtual SymbolTable& symbolTable() PURE;
 
+  /**
+   * Mark rejected stats as deleted by moving them to a different vector, so they don't show up
+   * when iterating over stats, but prevent crashes when trying to accesses references to them.
+   */
+  virtual void markCounterForDeletion(StatName name) PURE;
```

Comment:
The condition is still needed as stat names are duplicated across scopes, and we iterate over scopes to mark for deletion. However, I agree it can be jarring to see markCounterForDeletion take a StatName and not a counter.
Making the change.